### DISCUSSION TITLE
Weird error after latest update - arm-none-eabi-g++: error: missing argument to '-mcpu='

### DISCUSSION THREAD

**Discussion by LAPSyLAB** [stm32duino/Arduino_Core_STM32#2141] [Variants] [answered]
> Hello,

I get this error. It used to work before, not sure what changed after update. It seems like it doesn't find proper board anymore...

My procedure was to add support for the STM32H750B-DK board :

- add this section to boards.txt 
```
# STM32H750B-DK
Disco.menu.pnum.STM32H750B_DK=STM32H750B-DK
Disco.menu.pnum.STM32H750B_DK.upload.maximum_size=131072
Disco.menu.pnum.STM32H750B_DK.upload.maximum_data_size=884736
Disco.menu.pnum.STM32H750B_DK.build.board=GENERIC_H750XBHX
Disco.menu.pnum.STM32H750B_DK.build.product_line=STM32H750xx
Disco.menu.pnum.STM32H750B_DK.build.variant=STM32H7xx/H742X(G-I)H_H743X(G-I)H_H745X(G-I)H_H747X(G-I)H_H750XBH_H753XIH_H755XIH_H757XIH
```

- add ldscript.ld to latter map STM32H7xx...

It used to compile and basic LED/serial test was working... After update, I get this :

> FQBN: STMicroelectronics:stm32:Disco:opt=o0std,upload_method=swdMethod
> Using board 'Disco' from platform in folder: C:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0
> Using core 'arduino' from platform in folder: C:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0
> 
> "C:\\Users\\Uporabnik\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\tools\\STM32Tools\\2.2.1/win/busybox.exe" sh "C:\\Users\\Uporabnik\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\system/extras/prebuild.sh" "C:\\Users\\Uporabnik\\AppData\\Local\\Temp\\arduino\\sketches\\97652CF08BBA7A03557FFB9F5972A687" "D:\\OneDrive - Univerza v Ljubljani\\Delovni\\Workspaces\\Arduino_IDE\\sketch_stm32Duino_STM32H750" "C:\\Users\\Uporabnik\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0"
> Detecting libraries used...
> C:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\tools\xpack-arm-none-eabi-gcc\12.2.1-1.2/bin/arm-none-eabi-g++ -mcpu= -DVECT_TAB_OFFSET=0x0 -DUSE_FULL_LL_DRIVER -mthumb @C:\Users\Uporabnik\AppData\Local\Temp\arduino\sketches\97652CF08BBA7A03557FFB9F5972A687/sketch/build.opt -c -O0 -DNDEBUG -w -std=gnu++17 -ffunction-sections -fdata-sections -fno-threadsafe-statics --param max-inline-insns-single=500 -fno-rtti -fno-exceptions -fno-use-cxa-atexit -ID:\OneDrive - Univerza v Ljubljani\Delovni\Workspaces\Arduino_IDE\sketch_stm32Duino_STM32H750 -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/avr -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32 -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32/LL -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32/usb -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32/OpenAMP -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32/usb/hid -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32/usb/cdc -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Drivers/_HAL_Driver/Inc -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Drivers/_HAL_Driver/Src -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/ -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/ST/STM32_USB_Device_Library/Core/Inc -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/ST/STM32_USB_Device_Library/Core/Src -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/OpenAMP -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/OpenAMP/open-amp/lib/include -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/OpenAMP/libmetal/lib/include -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/OpenAMP/virtual_driver -w -x c++ -E -CC -D -DARDUINO=10607 -DARDUINO_GENERIC_H750XBHX -DARDUINO_ARCH_STM32 -DBOARD_NAME="GENERIC_H750XBHX" -DVARIANT_H="variant_GENERIC_H750XBHX.h" -DSTM32H750xx -DHAL_UART_MODULE_ENABLED -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\tools\CMSIS\5.7.0/CMSIS/Core/Include/ -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Drivers/CMSIS/Device/ST//Include/ -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Drivers/CMSIS/Device/ST//Source/Templates/gcc/ -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\tools\CMSIS\5.7.0/CMSIS/DSP/Include -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\tools\CMSIS\5.7.0/CMSIS/DSP/PrivateInclude -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\variants\STM32H7xx\H742X(G-I)H_H743X(G-I)H_H745X(G-I)H_H747X(G-I)H_H750XBH_H753XIH_H755XIH_H757XIH C:\Users\Uporabnik\AppData\Local\Temp\arduino\sketches\97652CF08BBA7A03557FFB9F5972A687\sketch\sketch_stm32Duino_STM32H750.ino.cpp -o nul
> Error while detecting libraries included by C:\Users\Uporabnik\AppData\Local\Temp\arduino\sketches\97652CF08BBA7A03557FFB9F5972A687\sketch\sketch_stm32Duino_STM32H750.ino.cpp
> 
> C:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\tools\xpack-arm-none-eabi-gcc\12.2.1-1.2/bin/arm-none-eabi-g++ -mcpu= -DVECT_TAB_OFFSET=0x0 -DUSE_FULL_LL_DRIVER -mthumb @C:\Users\Uporabnik\AppData\Local\Temp\arduino\sketches\97652CF08BBA7A03557FFB9F5972A687/sketch/build.opt -c -O0 -DNDEBUG -w -std=gnu++17 -ffunction-sections -fdata-sections -fno-threadsafe-statics --param max-inline-insns-single=500 -fno-rtti -fno-exceptions -fno-use-cxa-atexit -ID:\OneDrive - Univerza v Ljubljani\Delovni\Workspaces\Arduino_IDE\sketch_stm32Duino_STM32H750 -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/avr -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32 -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32/LL -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32/usb -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32/OpenAMP -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32/usb/hid -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32/usb/cdc -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Drivers/_HAL_Driver/Inc -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Drivers/_HAL_Driver/Src -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/ -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/ST/STM32_USB_Device_Library/Core/Inc -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/ST/STM32_USB_Device_Library/Core/Src -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/OpenAMP -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/OpenAMP/open-amp/lib/include -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/OpenAMP/libmetal/lib/include -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/OpenAMP/virtual_driver -w -x c++ -E -CC -D -DARDUINO=10607 -DARDUINO_GENERIC_H750XBHX -DARDUINO_ARCH_STM32 -DBOARD_NAME="GENERIC_H750XBHX" -DVARIANT_H="variant_GENERIC_H750XBHX.h" -DSTM32H750xx -DHAL_UART_MODULE_ENABLED -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\tools\CMSIS\5.7.0/CMSIS/Core/Include/ -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Drivers/CMSIS/Device/ST//Include/ -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Drivers/CMSIS/Device/ST//Source/Templates/gcc/ -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\tools\CMSIS\5.7.0/CMSIS/DSP/Include -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\tools\CMSIS\5.7.0/CMSIS/DSP/PrivateInclude -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\variants\STM32H7xx\H742X(G-I)H_H743X(G-I)H_H745X(G-I)H_H747X(G-I)H_H750XBH_H753XIH_H755XIH_H757XIH C:\Users\Uporabnik\AppData\Local\Temp\arduino\sketches\97652CF08BBA7A03557FFB9F5972A687\sketch\SrcWrapper.cpp -o nul
> Error while detecting libraries included by C:\Users\Uporabnik\AppData\Local\Temp\arduino\sketches\97652CF08BBA7A03557FFB9F5972A687\sketch\SrcWrapper.cpp
> Generating function prototypes...
> C:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\tools\xpack-arm-none-eabi-gcc\12.2.1-1.2/bin/arm-none-eabi-g++ -mcpu= -DVECT_TAB_OFFSET=0x0 -DUSE_FULL_LL_DRIVER -mthumb @C:\Users\Uporabnik\AppData\Local\Temp\arduino\sketches\97652CF08BBA7A03557FFB9F5972A687/sketch/build.opt -c -O0 -DNDEBUG -w -std=gnu++17 -ffunction-sections -fdata-sections -fno-threadsafe-statics --param max-inline-insns-single=500 -fno-rtti -fno-exceptions -fno-use-cxa-atexit -ID:\OneDrive - Univerza v Ljubljani\Delovni\Workspaces\Arduino_IDE\sketch_stm32Duino_STM32H750 -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/avr -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32 -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32/LL -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32/usb -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32/OpenAMP -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32/usb/hid -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/stm32/usb/cdc -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Drivers/_HAL_Driver/Inc -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Drivers/_HAL_Driver/Src -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/ -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/ST/STM32_USB_Device_Library/Core/Inc -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/ST/STM32_USB_Device_Library/Core/Src -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/OpenAMP -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/OpenAMP/open-amp/lib/include -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/OpenAMP/libmetal/lib/include -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Middlewares/OpenAMP/virtual_driver -w -x c++ -E -CC -D -DARDUINO=10607 -DARDUINO_GENERIC_H750XBHX -DARDUINO_ARCH_STM32 -DBOARD_NAME="GENERIC_H750XBHX" -DVARIANT_H="variant_GENERIC_H750XBHX.h" -DSTM32H750xx -DHAL_UART_MODULE_ENABLED -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\tools\CMSIS\5.7.0/CMSIS/Core/Include/ -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Drivers/CMSIS/Device/ST//Include/ -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\system/Drivers/CMSIS/Device/ST//Source/Templates/gcc/ -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\tools\CMSIS\5.7.0/CMSIS/DSP/Include -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\tools\CMSIS\5.7.0/CMSIS/DSP/PrivateInclude -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino -IC:\Users\Uporabnik\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\variants\STM32H7xx\H742X(G-I)H_H743X(G-I)H_H745X(G-I)H_H747X(G-I)H_H750XBH_H753XIH_H755XIH_H757XIH C:\Users\Uporabnik\AppData\Local\Temp\arduino\sketches\97652CF08BBA7A03557FFB9F5972A687\sketch\sketch_stm32Duino_STM32H750.ino.cpp -o C:\Users\Uporabnik\AppData\Local\Temp\3931218157\sketch_merged.cpp
> arm-none-eabi-g++: error: missing argument to '-mcpu='
> exit status 1
> 
> Compilation error: exit status 1

Any advice what is wrong ?

Thanks.
[ldscript.zip](https://github.com/stm32duino/Arduino_Core_STM32/files/12829537/ldscript.zip)

**Comment by fpistm** [comment, marked as answer]
Hi @LAPSyLAB 
that is normal you don't properly define your board in the boards.txt.
Compare to this entry:
https://github.com/stm32duino/Arduino_Core_STM32/blob/61a1680ef96871a38672b5d9ea57d7a9df9b3141/boards.txt#L1036-L1047

![image](https://github.com/stm32duino/Arduino_Core_STM32/assets/20641798/38fd8a72-4ef0-4209-af5c-1f96cddc3ee5)
It seems you try to add the generic support, in that case, simply add it to the generic section for H7, here the missing entries are common for all H7. In the Disco section each entry have to define the .mcu, .fpu, .float-abi and .series.

https://github.com/stm32duino/Arduino_Core_STM32/blob/61a1680ef96871a38672b5d9ea57d7a9df9b3141/boards.txt#L7361-L7370

**Reply by fpistm** [reply]
For Ref: https://www.stm32duino.com/viewtopic.php?p=13209#p13209
